specs/CameraCaptureUI/CameraCaptureUI.md

```diff
@@ -0,0 +1,387 @@
+Your app may fail Windows Application Certification Kit certification when submitted to 
+the Store if you specify the webcam or microphone capabilities when using CameraCaptureUI.
+
+
+
+## CameraCaptureUI Properties
+
+| Name | Description | Value |
+|-|-|-|
+|PhotoSettings| Provides settings for capturing photos. The settings include aspect ratio, image size, format, resolution, and whether or not cropping is allowed by the user interface (UI).|CameraCaptureUIPhotoCaptureSettings (An object containing settings for capturing photos) |
+|VideoSettings| Provides settings for capturing videos. The settings include format, maximum resolution, maximum duration, and whether or not to allow trimming.| CameraCaptureUIVideoCaptureSettings (An object that provides settings for capturing videos) |
+
+
+
+
+## CameraCaptureUI Methods
+
+| Name | Description | Parameters | Returns |
+|-|-|-|-|
+| CaptureFileAsync(CameraCaptureUIMode) | Launches the CameraCaptureUI user interface.| __mode__ CameraCaptureUIMode | __IAsyncOperation< StorageFile >__ |
+
+
+
+
+## CameraCaptureUIPhotoCaptureSettings Class
+Provides settings for capturing photos with CameraCaptureUI. 
+The settings include aspect ratio, image size, format, resolution, 
+and whether or not cropping is allowed by the user interface (UI).
+
+## CameraCaptureUIPhotoCaptureSettings Properties
+
+| Name | Description | Value |
+|-|-|-|
+| AllowCropping | Determines whether photo cropping will be enabled in the user interface for capture a photo.| Boolean |
+| CroppedAspectRatio | The aspect ratio of the captured photo.| Windows.Foundation.Size |
+| CroppedSizeInPixels |The exact size in pixels of the captured photo.| Windows.Foundation.Size |
+| Format |Determines the format that captured photos will be stored in.| CameraCaptureUIPhotoFormat |
+| MaxResolution | Determines the maximum resolution the user will be able to select. | CameraCaptureUIMaxPhotoResolution |
+
+
+
+
+## CameraCaptureUIVideoCaptureSettings Class
+Provides settings for capturing videos. 
+The settings include format, maximum resolution, maximum duration, and whether or not to allow trimming.
+
+## CameraCaptureUIVideoCaptureSettings Properties
+| Name | Description | Value |
+|-|-|-|
+| AllowTrimming | Determines whether the user can trim the captured video.| Boolean |
+| Format | Determines the format that captured videos will be stored in.| CameraCaptureUIVideoFormat |
+| MaxDurationInSeconds | The maximum duration of the video that the user can capture.| Single |
+| MaxResolution | Determines the maximum resolution the user will be able to select.| CameraCaptureUIMaxVideoResolution |
+
+
+## CameraCaptureUIMode enum
+
+Determines whether the user interface for capturing from the attached camera allows capture of photos, 
+videos, or both photos and videos.
+
+| Name | Value | Description |
+|-|-|-|
+| PhotoOrVideo | 0 | The user interface allows capture of photos and videos. |
```

Comment:
Which value is the default if the developer doesn't explicitly specify one?  The same question applies to other fields such as PhotoFormat and VideoFormat.

Reply:
Generally agree, but actually none of these have defaults. This one (`CameraCaptureUIMode`) is a required parameter to `CaptureFileAsync`. The others are properties on Settings classes you get from `CameraCaptureUI.PhotoSettings` and `VideoSettings`, and so the initial value is a feature of those APIs.

Reply:
So, the CameraCapture API will function even if you don't specify these properties. It only requires that you specify the mode, such as Photo, Video, or PhotoOrVideo. I agree with Mike's point.

Reply:
What are the default values on those sub-classes, though?  If I fail to specify properties do I get the highest possible quality or the lowest possible quality?  Or something in the middle?

Reply:
I'm arguing that it would be inappropriate to document that in this case. The values you get from `CameraCaptureUI.PhotoSettings` is a feature of that API. Maybe the value you get back is a hard-coded value, or maybe it's something that the user sets in the Camera app settings.

Reply:
I can buy that.  Perhaps it would also vary based on hardware (e.g. devices with more RAM default to higher resolution or something like that).